### ISSUE TITLE
Keyboard navigation not working for urlbar suggestions

### ISSUE THREAD

**Issue by jkup** [brave/browser-laptop#4218]
**Did you search for similar issues before submitting this one?**
Yes.

**Describe the issue you encountered:**

There are a few issues surrounding keyboard navigation and our URLbar. Ideally for accessibility everything that can be done in Brave should be able to be done with the keyboard only.

Start typing in the URLbar until search results appear. Key down until you find a result you like and then hit tab to select it (everything works well up until now). Notice that you can't continue typing because the search result is selected and will be replaced (this is ok, not ideal) and you can't use the arrow keys to deselect the result and continue typing (this is the real bug)

**Expected behavior:**

You should be able to use tab to select a result, hit the right arrow and then continue typing.
- Platform (Win7, 8, 10? macOS? Linux distro?):

macOS
- Brave Version:

0.12.1
